template.tpl

```diff
@@ -436,7 +588,209 @@ ___WEB_PERMISSIONS___
+      return function(item) {
+        assertThat(item).isNotNull();
+      };
+    });
+
+    mock('setInWindow', (key, val) => {
+      if (key === 'rdt') rdt = val;
+    });
+
+    // Call runCode to run the template's code.
+    runCode(mockData);
+
+    // Verify that the tag finished successfully.
+    assertApi('setInWindow').wasCalled();
+    assertApi('gtmOnSuccess').wasCalled();
+- name: Test enabledFirstPartyCookies false
+  code: |-
+    mockData.enableFirstPartyCookies = false;
+
+    let isSet = false;
+
+    mock('copyFromWindow', key => {
+      if (key === 'rdt') return function() {
+         if (arguments[0] === 'disableFirstPartyCookies') {
+           isSet = true;
+         }
+      };
+    });
+
+    // Call runCode to run the template's code.
+    runCode(mockData);
+
+    // Verify that the tag finished successfully.
+    assertThat(isSet, 'disableFirstPartyCookies not set even though first party cookies are disabled').isTrue();
+    assertApi('gtmOnSuccess').wasCalled();
+- name: Test enabledFirstPartyCookies true
+  code: |-
+    mockData.enableFirstPartyCookies = true;
+
+    mock('copyFromWindow', key => {
+      if (key === 'rdt') return function() {
+         if (arguments[0] === 'disableFirstPartyCookies') fail('disableFirstPartyCookies called even though first party cookies are enabled');
+      };
+    });
+
+    // Call runCode to run the template's code.
+    runCode(mockData);
+
+    // Verify that the tag finished successfully.
+    assertApi('gtmOnSuccess').wasCalled();
+- name: Test Advanced Matching
+  code: |
+    mockData = {
+      id: "t2_123",
+      event_type: "PageVisit",
+      enableFirstPartyCookies: true,
+      advancedMatching: true,
+      advancedMatchingParams: [
+        {name: 'email', value: 'alice@example.com'},
+        {name: 'aaid', value: 'cdda802e-fb9c-47ad-9866-0794d394c912'},
+        {name: 'idfa', value: 'EA7583CD-A667-48BC-B806-42ECB2B48606'}
+      ],
+    };
+
+    const expected = {
+      email: 'alice@example.com',
+      aaid: 'cdda802e-fb9c-47ad-9866-0794d394c912',
+      idfa: 'EA7583CD-A667-48BC-B806-42ECB2B48606',
+      integration: 'gtm'
+    };
+
+    mock('copyFromWindow', key => {
+      if (key === 'rdt') return function() {
+         if (arguments[0] === 'init') {
+          assertThat(arguments[2], 'Advanced matching parameters incorrect').isEqualTo(expected);
+        }
+      };
+    });
+
+    // Call runCode to run the template's code.
+    runCode(mockData);
+
+    // Verify that the tag finished successfully.
+    assertApi('makeTableMap').wasCalledWith(mockData.advancedMatchingParams, 'name', 'value');
+    assertApi('gtmOnSuccess').wasCalled();
+- name: Test Event Metadata
+  code: |-
+    mockData = {
+      id: "t2_potato",
+      event_type: "Purchase",
+      enableFirstPartyCookies: true,
+      eventMetadataParams: [
+        {name: "itemCount", value: "1"},
+        {name: "value", value: "1000"},
```

Comment:
The pixel is expecting `itemCount` and `value` to be numbers. For consistency can we make these numbers here?